Tests/SwiftStdlibTests/MutableCollectionTests.swift

```diff
@@ -33,4 +33,37 @@ final class MutableCollectionTests: XCTestCase {
         array2.sort(by: \String.first, with: optionalCompare)
         XCTAssertEqual(array2, ["Bryant", "James", "Wade", ""])
     }
+
+    func testAssignToAll() {
+        struct TestStruct : ExpressibleByIntegerLiteral, CustomStringConvertible, Equatable {
+            var testField: Int = 0
+
+            typealias IntegerLiteralType = Int
+
+            init(integerLiteral value: Int) {
+                self.testField = value
+            }
+
+            // added for being able to perform assert equal
+            static func ==(lhs: TestStruct, rhs: TestStruct) -> Bool {
```

Comment:
Since this is a struct, this equatable is auto syntetized by the compiler, so you don't need to actually implement this :)

Reply:
I firstly did try that way but I got a compiler error of this kind
`Error:(56, 9) global function 'XCTAssertEqual(_:_:_:file:line:)' requires that 'TestStruct' conform to 'Equatable'`

Reply:
That's weird ... the compiler should generate the syntetized equals if you add the `Equatable` conformance and all stored members are equatable.
I'll check that as soon as possible :))

Reply:
> That's weird ... the compiler should generate the syntetized equals if you add the `Equatable` conformance and all stored members are equatable.
> I'll check that as soon as possible :))

@LucianoPAlmeida, it was my bad, it works as you said, sorry for this